### ISSUE TITLE
Pull request method for correcting Cologne dictionaries

### ISSUE THREAD

**Issue by drdhaval2785** [sanskrit-lexicon/csl-corrections#31]
This issue is dedicated to give a step by step visual guide to a user wanting to submit batch of error correction via git and github on a regular manner.

For this example, we would make the following correction

Dictionary - MW72
headword - atipraRAS
See https://www.sanskrit-lexicon.uni-koeln.de/scans/MW72Scan/2014/web/webtc/indexcaller.php?key=atipraRAS&input=slp1&output=SktDevaUnicode
This gives the following result.
```
अतिप्रणाश् [Cologne record ID=737] [Printed book page 0015-a]अतिप्रणाश् ati-pra-ṇaś (rt. naś), cl. 4. P.-ṇaśyati, -ṇaśitum or -ṇaṃṣṭum, to be deprivedof (with acc.).
```

When it is crosschecked with the scan at https://www.sanskrit-lexicon.uni-koeln.de/scans/awork/apidev/servepdf.php?dict=MW72&key=atipraRAS , it gives the following.
![capture](https://user-images.githubusercontent.com/5755132/103498431-c7c4de80-4e6a-11eb-9f85-fb1e046d64a7.PNG)

On examining the scan, it is verified that the headword needs correction from `atipraRAS` to `atipraRaS`.

The following issue would make a step by step visual instruction to make such a correction and let the maintainers of Cologne dictionaries know about it by a pull request.

**Comment by drdhaval2785** [comment]
# Prerequisites

1. You should have git installed in your computer. Read https://github.com/sanskrit-lexicon/csl-corrections/issues/23 for details.
2. You should have a github account.

**Comment by drdhaval2785** [comment]
# Step 1

1. Sign in to your github account with your username and password.

**Comment by drdhaval2785** [comment]
# Step 2

1. Go to https://github.com/sanskrit-lexicon/csl-orig
2. Click on 'fork' button on right side of the page, as shown in the figure below.

![capture](https://user-images.githubusercontent.com/5755132/103498797-f1cad080-4e6b-11eb-8a8f-f90a11d49a18.PNG)

**Comment by drdhaval2785** [comment]
# Step 3

1. Open your fork of csl-orig (In my case https://github.com/drdhaval2785/csl-orig).
2. Click on 'Code' button on the right side.
![capture](https://user-images.githubusercontent.com/5755132/103499171-2ee39280-4e6d-11eb-8c38-23d4575f3d42.PNG)

3. Select the line in 'HTTPS' tab, and copy it by Ctrl-C or clicking on clipboard icon.
![capture](https://user-images.githubusercontent.com/5755132/103499238-605c5e00-4e6d-11eb-8422-a5d3ff72d856.PNG)

**Comment by drdhaval2785** [comment]
# Step 4

1. Create a folder named 'cologne' at a convenient place. (I have created it at Desktop/cologne).
2. Open 'Git Bash'.

![capture](https://user-images.githubusercontent.com/5755132/103499660-db724400-4e6e-11eb-8a06-a455c976eafe.PNG)

3. Type `cd /path/to/directory` and press ENTER. In my case it would be `cd Desktop/cologne`.
 
![capture](https://user-images.githubusercontent.com/5755132/103499747-3015bf00-4e6f-11eb-9d8c-070eea5c8fc0.PNG)

4. Type `git clone https://github.com/{USERNAME}/csl-orig.git` and press ENTER. In my case, it would be `git clone https://github.com/drdhaval2785/csl-orig.git`. Make sure to type your own username. The line after `git clone ` would be the same as what we had copied from step 3.3. Wait for the process to end. At the end, the screen would show something like this.

![capture](https://user-images.githubusercontent.com/5755132/103499995-0f019e00-4e70-11eb-9575-b0bc5ff756b7.PNG)

5. This means that you now have your own copy of csl-orig repository in your computer.

**Comment by drdhaval2785** [comment]
# Step 5
1. Open your file editor, and reach to the dictionary which you want to edit. Path would be `cologne/csl-orig/v02/{DICT}`. In our example, we want to edit the mw72 dictionary. Therefore, we will go to `cologne/csl-orig/v02/mw72` folder.
![capture](https://user-images.githubusercontent.com/5755132/103500164-8e8f6d00-4e70-11eb-8a07-b7e81c5cf404.PNG)

2. Open DICT.txt file in the text editor of your choice. In our case, it would be mw72.txt file. I am using notepad++. You may use notepad++, emeditor, emacs. Simple notepad usually hangs. Therefore, its use is not recommended.

![capture](https://user-images.githubusercontent.com/5755132/103500304-ed54e680-4e70-11eb-9540-c54a46c16939.PNG)

3. Search for the entry you want to correct. In our case, we want to correct `atipraRAS` entry. 
![capture](https://user-images.githubusercontent.com/5755132/103500402-35740900-4e71-11eb-9927-2bb2b2d98faf.PNG)

4. Make necessary corrections in the entry.
OLD ENTRY
```
<L>737<pc>0015-a<k1>atipraRAS<k2>atipraRAS
{#atipraRAS#}¦ {%ati-pra-ṇaś%} (rt. {%naś%}), cl. 4. P.
<div n="lb"/>{%-ṇaśyati, -ṇaśitum%} or {%-ṇaṃṣṭum,%} to be deprived
<div n="lb"/>of (with acc.).
<LEND>
```
CORRECTED ENTRY
```
<L>737<pc>0015-a<k1>atipraRaS<k2>atipraRaS
{#atipraRaS#}¦ {%ati-pra-ṇaś%} (rt. {%naś%}), cl. 4. P.
<div n="lb"/>{%-ṇaśyati, -ṇaśitum%} or {%-ṇaṃṣṭum,%} to be deprived
<div n="lb"/>of (with acc.).
<LEND>
```

5. Save the file.

**Comment by drdhaval2785** [comment]
# Step 6

1. Open 'git bash'
2. Reach the csl-orig repository by `cd /path/to/repository` e.g. `cd Desktop/cologne/csl-orig`.
3. Type `git status`. This will show you the changed file in red colour.
![capture](https://user-images.githubusercontent.com/5755132/103500709-e5e20d00-4e71-11eb-8ce2-2662543f3d1e.PNG)

4. Type `git diff`. This will show you the changes you made to the file.

![capture](https://user-images.githubusercontent.com/5755132/103500782-1de95000-4e72-11eb-9bbb-1da14fad90b2.PNG)

points 3 and 4 are just for sanity check to see if you are actually sending only the desired corrections or not.

5. Type `git add .`
6. Type `git commit -m "CORRECTION MESSAGE"`. In our case, I am sending the message 'MW72 atipraRAS->atipraRAS'. So the command would be `git commit -m "MW72 atipraRAS->atipraRaS"`
![capture](https://user-images.githubusercontent.com/5755132/103500984-ddd69d00-4e72-11eb-9ec6-776469f51b41.PNG)

![image](https://user-images.githubusercontent.com/5755132/103501031-fc3c9880-4e72-11eb-92ef-6cb9c2e1ec19.png)

7. Type `git push origin master`. Type your username and password when prompted.
![capture](https://user-images.githubusercontent.com/5755132/103501150-61908980-4e73-11eb-9e0e-b7076ac074df.PNG)

8. Wait for the process to end. You will see something like this.
![capture](https://user-images.githubusercontent.com/5755132/103501336-e11e5880-4e73-11eb-961c-157358110986.PNG)

You have successfully uploaded the change to your copy of csl-orig i.e. drdhaval2785/csl-orig.
Now it is time to notify that change to sanskrit-lexicon/csl-orig via a pull request.
That would be handled in the next step.

**Comment by drdhaval2785** [comment]
# Step 7

1. Open https://github.com/{USERNAME}/csl-orig . In our case, it is https://github.com/drdhaval2785/csl-orig.
2. Look at the highlighted line. It will tell you that you have committed something which is not yet incorporated at sanskrit-lexion/csl-orig.

![capture](https://user-images.githubusercontent.com/5755132/103501538-7cafc900-4e74-11eb-93a3-bf2d2da5c553.PNG)

3. Click on 'Pull Request' button on the right side.
![capture](https://user-images.githubusercontent.com/5755132/103501590-a79a1d00-4e74-11eb-8455-cbdd75b67ead.PNG)

4. Click on 'Create Pull Request' button.
![image](https://user-images.githubusercontent.com/5755132/103501685-f1830300-4e74-11eb-841d-85e7ad6f2d1c.png)

5. Once again, click on 'Create Pull Request' button. You can also write some message if you wish to, but it is optional.

![capture](https://user-images.githubusercontent.com/5755132/103501750-27c08280-4e75-11eb-9e16-3a6b89044e12.PNG)

6. You have now successfully created a pull request. 

![capture](https://user-images.githubusercontent.com/5755132/103501856-7cfc9400-4e75-11eb-9ccf-fe5d2ffc927a.PNG)

7. Maintainers of sanskrit-lexicon/csl-orig repository can see your pull request. If found OK, they can merge this pull request, and your change would be available in sanskrit-lexicon/csl-orig repository.

**Comment by gasyoun** [comment]
> This issue is dedicated to give a step by step visual guide to a user wanting to submit error correction via git and github.

I would say a batch of errors on a regular manner.

**Comment by drdhaval2785** [comment]
@sanskritisampada would also be a candidate. Maybe Usha also.

**Comment by gasyoun** [comment]
> If found OK, they can merge this pull request, and your change would be available in sanskrit-lexicon/csl-orig repository.

In a week or so.

> @sanskritisampada would also be a candidate. Maybe Usha also.

Let us try the Desktop approach with Usha, it does get a bit harsh with a console.

**Comment by drdhaval2785** [comment]
Congratulations @AnnaRybakovaT for your first pull request. 
Now you are familiar with the functioning of pull requests. You can now send batches of 20 or so corrections in a single pull request. 

@funderburkjim Are you OK with this method of correction submission?
In case there are any discussions (think of notes after semicolon in our old system), we can discuss the same in 'conversation' portion of pull request.

**Comment by funderburkjim** [comment]
@drdhaval2785 

A couple of years ago, you urged that I trade in my horse and buggy for a shiny new git-mobile.

After some growing pains,  I finally came to love it.   

Now, you're pointing out the new pull-request gadget which I have never made use of.

Probably I'll come to love it, also.  Willing to give a try.

I'm sure some new questions/concerns will initially arise.

**Comment by funderburkjim** [comment]
## Notification question

I have no email notification regarding either the pull request that Anna made,  or your merge of the
pull request.   

The only way I know that something happened to csl-orig is by looking at it in Github.

**Comment by funderburkjim** [comment]
## multiple batch question

Suppose Anna or Sampada do several batches of corrections.

In detail, suppose Anna has done a fork of csl-orig, cloned it to her local computer,  made 
Batch 1 of corrections to MW local csl-orig, and made a pull request.

Suppose she then makes a comment in a Github issue to notify me of this pull request.

Next day she wants to start on Batch  2.   Where does she start?   Does she do another fork
of csl-orig?   How does that work ?

**Comment by gasyoun** [comment]
> A couple of years ago, you urged that I trade in my horse and buggy for a shiny new git-mobile.

And now he offers you Ford Model T. It will add millions of updates to my github panel, still.

**Comment by drdhaval2785** [comment]
> Notification question
I received the usual mail notification.
You did not get the notification, because you are not watching the csl-orig repository. Check https://github.com/sanskrit-lexicon/csl-orig/watchers . I am the only member who is watching this repository.

You may click on the 'Watch' button on right upper corner, and you should receive all the notifications.

**Comment by drdhaval2785** [comment]
> multiple batch question
 Next day she wants to start on Batch 2. Where does she start? Does she do another fork
of csl-orig? How does that work ?


She can continue to work in her local copy. She can create changes, `git add .`, `git commit -m "blah blah"`, `git push origin master` to her https://github.com/AnnaRybakovaT/csl-orig
Once her fork has those changes, she can once again crete a pull request.

**Comment by drdhaval2785** [comment]
# Upadate your fork (Background)

Based on https://stackoverflow.com/questions/20984802/how-can-i-keep-my-fork-in-sync-without-adding-a-separate-remote/21131381#21131381

There are cases where your fork https://github.com/drdhaval2785/csl-orig does not contain the changes made to the original repository https://github.com/sanskrit-lexicon/csl-orig .
It is most probably due to some other contributor submitting a change on https://github.com/sanskrit-lexicon/csl-orig . 

A similar case presented itself in our exercise.

@AnnaRybakovaT submitted a correction in mw72.txt via https://github.com/sanskrit-lexicon/csl-orig/pull/432 and @funderburkjim made various changes to ap90.txt file like https://github.com/sanskrit-lexicon/csl-orig/commit/ccd474079b2c109a7460eccaacefb03596bbee94 .

These two changes are existing on https://github.com/sanskrit-lexicon/csl-orig , but are absent from my fork https://github.com/drdhaval2785/csl-orig . 
To get the maximum out of the updates on the original repository, I need to (1) Update my fork https://github.com/drdhaval2785/csl-orig to have the changes mentioned above and (2) Update my local files to have those changes.

The next comments would give a visual guide for the same

NOTE - It is advisable to have the fork synced and local files updated before starting any new corrections.

**Comment by drdhaval2785** [comment]
# Update Forked Repo (Step 1)

1. Open your fork on Github - https://github.com/{USERNAME}/csl-orig e.g. https://github.com/drdhaval2785/csl-orig

2. This will show whether your fork is behind the master or not.
![capture](https://user-images.githubusercontent.com/5755132/103612059-a129a500-4f49-11eb-94a2-5a32db3dc703.PNG)

3. If it is behind the master, click on 'Pull Request' as highlighted below.
![capture](https://user-images.githubusercontent.com/5755132/103612153-ccac8f80-4f49-11eb-8df6-f976be5a9c36.PNG)

4. Click on 'switching the base'
![capture](https://user-images.githubusercontent.com/5755132/103612216-f5cd2000-4f49-11eb-8142-ba5ec41a08d4.PNG)

5. You should be able to see (1) commits which are different between your repo and original repo. (2) Difference in files. They are marked with 1 and 2 in the below image. Check the differences. Once you are OK with the changes, move to the next point.
![capture](https://user-images.githubusercontent.com/5755132/103612413-6411e280-4f4a-11eb-8fba-5ee78a355b37.PNG)

6. Click on 'Create Pull Request' 
![capture](https://user-images.githubusercontent.com/5755132/103612519-9facac80-4f4a-11eb-8cad-9d6486e03369.PNG)

7. You shall see a window like the one given below. (1) If there are no clashes, there will be a message `Able to merge. These branches can be automatically merged.`. It is safe to merge this pull request. If there is any other message, STOP. (2) Write some useful name for the pull request. I have written `pull ap90.txt and mw72.txt changes`. (3) Click on 'Create Pull Request'.
![capture](https://user-images.githubusercontent.com/5755132/103612743-15b11380-4f4b-11eb-9dea-f30cc8e5c5fd.PNG)

8. Click on `Merge Pull Request` and `Create a merge commit`
![capture](https://user-images.githubusercontent.com/5755132/103612876-5c9f0900-4f4b-11eb-9179-60857e2f00f0.PNG)

9. Click on 'Confirm merge'.
![capture](https://user-images.githubusercontent.com/5755132/103612932-7dfff500-4f4b-11eb-94d3-3e0601b8e3e8.PNG)

10. The pull request is now merged in https://github.com/drdhaval2785/csl-orig
![capture](https://user-images.githubusercontent.com/5755132/103613016-ab4ca300-4f4b-11eb-8f66-0747c7612986.PNG)

Now my forked repository has all the changes made to https://github.com/sanskrit-lexicon/csl-orig

11. Verify that your fork has desired changes (Not required. This is for extra precaution)
![capture](https://user-images.githubusercontent.com/5755132/103613520-af2cf500-4f4c-11eb-8fbd-410c9a712991.PNG)

And click on the commit. This shows the differences which are added via this updation process. Everything seems fine. My fork now has all the changes made by original repository.

![capture](https://user-images.githubusercontent.com/5755132/103613858-5ad64500-4f4d-11eb-9048-a8181182923d.PNG)

**Comment by drdhaval2785** [comment]
# Update Forked Repo (Step 2)

In this step, we will update our local files based on updated https://github.com/drdhaval2785/csl-orig via Step 1.

1. Open `Git Bash`.
2. `cd /path/to/repository`
3. `git status` . If there are any changes which are not committed, it will be shown. Then you should do `git add .` and `git commit -m "commit message"`. If there are no uncommitted changes, you will see a message `Your branch is up to date with 'origin/master'. nothing to commit, working tree clean`. Then proceed to step 4.
4. `git pull origin master`

![capture](https://user-images.githubusercontent.com/5755132/103616156-e8b42f00-4f51-11eb-8792-b5d36ceaa205.PNG)

**Comment by drdhaval2785** [comment]
@funderburkjim 
I hope that above messages will help @AnnaRybakovaT and anyone else working with forked csl-orig repository to update his / her repository when there are new changes on sanskrit-lexicon/csl-orig repository.
Anna may give it a try, and update her fork to be up to date with changes in ap90.txt made by Jim.

**Comment by gasyoun** [comment]
> Anna may give it a try, and update her fork to be up to date with changes in ap90.txt made by Jim.

Documented well, but would not say it's easy to grasp, let us see, until now Anna never failed to, thanks.

**Comment by AnnaRybakovaT** [comment]
> Anna may give it a try, and update her fork to be up to date with changes in ap90.txt made by Jim.

OK. I will try.

**Comment by AnnaRybakovaT** [comment]
I have started the Step 1.
Until 1.7 everything was fine but in 1.8 I stopped since after clicking on "Create a merge commit" - nothing happening

![image](https://user-images.githubusercontent.com/74726889/103678819-1a021e80-4f95-11eb-96d9-8f961d4e3e11.png)

**Comment by funderburkjim** [comment]
> You may click on the 'Watch' button.

Done Watching 'all activity' for csl-orig.  Thanks.

**Comment by funderburkjim** [comment]
Instructions on keeping forked csl-orig in sync with master csl-orig very useful.  

But there are some edge cases in the 'merge' process that could require further attention.  'Git' is very complete -- after all
it was developed by Linus Torvalds for maintaining Linux,  which has probably thousands of collaborators;  so Git can handle
basically any situation.  But there do arise situations  (such as when person A and person B make a change to the same line
in a file)  whose resolution is difficult.    But these are thankfully rare;  and we don't have to completely understand at the
beginning every single case.  

> In regard to 'Update Forked Repo (Step 2).  items 3 and 4' ([comment above](https://github.com/sanskrit-lexicon/csl-corrections/issues/31#issuecomment-754444135))

Step 2 is about pulling the merged fork to our local machine.  I actually think it does not matter whether 3 is done before 4.
In other words, I think we could just as well do the 'git pull' first.   Then sometime later do git add, git commit.

**Comment by funderburkjim** [comment]
https://github.com/AnnaRybakovaT/csl-orig  (Anna's fork of sanskrit-lexicon/csl-orig)   shows
`This branch is 5 commits behind sanskrit-lexicon:master.`

**Comment by drdhaval2785** [comment]
@AnnaRybakovaT 
n step 1.8, you needed to `Click on Merge Pull Request and Create a merge commit`.
When you click on Merge Pull Request, there will be a drop down `Create a merge commit`. You need to select that in dropdown.

**Comment by gasyoun** [comment]
> But there do arise situations (such as when person A and person B make a change to the same line
in a file) whose resolution is difficult. 

Rare is the word.

**Comment by AnnaRybakovaT** [comment]
> When you click on Merge Pull Request, there will be a drop down `Create a merge commit`. You need to select that in dropdown.

Yes, I see and click "Create a merge commit" - but no result.

![image](https://user-images.githubusercontent.com/74726889/103756203-18316d00-5020-11eb-8ee8-8b463fe014f6.png)

**Comment by AnnaRybakovaT** [comment]
As well today I see the new 3 commits

![image](https://user-images.githubusercontent.com/74726889/103757571-0355d900-5022-11eb-9ed2-704836b66041.png)

**Comment by drdhaval2785** [comment]
Hi Anna,
Your pull request was created successfully at https://github.com/AnnaRybakovaT/csl-orig/pull/1. Now you need to merge the same.

**Comment by drdhaval2785** [comment]
![Uploading Screenshot_20210106-170559_Chrome.jpg…]()

**Comment by drdhaval2785** [comment]
![Screenshot_20210106-170559_Chrome](https://user-images.githubusercontent.com/5755132/103764511-994d2c00-5041-11eb-93be-3018d4d889f6.jpg)

**Comment by drdhaval2785** [comment]
As I do not have write access to your repository, I am not getting any facility to merge the same. You must be getting some button to merge the pull request at the bottom (after the place to write message).

**Comment by AnnaRybakovaT** [comment]
> You must be getting some button to merge the pull request at the bottom (after the place to write message).

Unfortunately, I don't see anything like this

![image](https://user-images.githubusercontent.com/74726889/103766355-c2fd5780-502f-11eb-8d0a-1a7c0bab4f11.png)

**Comment by AnnaRybakovaT** [comment]
I found solution!!!!
I have to click one more time on "Merge pull request" (after activation "create a merge commit")

**Comment by AnnaRybakovaT** [comment]
So the Step 1 has done. I will make the Step 2 in a while.

**Comment by AnnaRybakovaT** [comment]
Dear all,

The Step 2 is done as well.
There was a message: Your branch is up to date with 'origin/master'. nothing to commit, working tree clean. So after "git status" I put straightly "git pull origin master"

![image](https://user-images.githubusercontent.com/74726889/103806257-d0353900-5065-11eb-86fd-498cda2bc07b.png)

**Comment by gasyoun** [comment]
> The Step 2 is done as well.

You've become a coder by now, I congratulate you.

**Comment by drdhaval2785** [comment]
# Updating a forked repo (When both ahead and behind)

Recently @AnnaRybakovaT referred to a situation where her repo was X commits ahead and Y commits behind the original repository. 
This issue tries to replicate similar situation for https://github.com/drdhaval2785/csl-orig and try to find a solution.
I have made a small correction regarding a single entry in PE in my fork. See https://github.com/sanskrit-lexicon/csl-corrections/issues/35 for error corrected.
I have pushed that change to https://github.com/drdhaval2785/csl-orig

1. Check the status of your repository (whether ahead or behind of sanskrit-lexicon/csl-orig). In my  case, it is 1 commit ahead and 3 commits behind the master.
![capture](https://user-images.githubusercontent.com/5755132/104094961-2d292d00-52ba-11eb-95a4-b2a8b3f8d968.PNG)

2. Click on 'Pull Request'
![image](https://user-images.githubusercontent.com/5755132/104094989-61045280-52ba-11eb-8f48-9b19341b0826.png)

3. You will see 'base repository' i.e. the repository on which the pull request is to be made, and 'head repository' i.e. the repository from which the pull request is to be made. In default case, it mentions that the 'base repository' i.e. sanskrit-lexicon/csl-orig and 'head repository' i.e. drdhaval2785/csl-orig differ in case of 2 lines of pe.txt shown below. 

![image](https://user-images.githubusercontent.com/5755132/104095064-d7a15000-52ba-11eb-9ad4-54301f510cc1.png)

4. We want to bring 3 commits already there on sanskrit-lexicon/csl-orig to drdhaval2785/csl-orig first. So, we change the 'base repository' to 'drdhaval2785/csl-orig' and 'head repository' to 'sanskrit-lexicon/csl-orig'. This is a bit tricky. Please see the steps closely.
5. Select 'drdhaval2785/csl-orig' in the 'base repository'
![image](https://user-images.githubusercontent.com/5755132/104095194-74fc8400-52bb-11eb-9c8e-b62983c08308.png)

6. You will now see a blank kind of screen (because you are comparing drdhaval2785/csl-orig to drdhaval2785/csl-orig i.e. no change would be there)
![image](https://user-images.githubusercontent.com/5755132/104095221-affeb780-52bb-11eb-8d99-fd3cf4d80758.png)

7. Click on 'compare across forks'
![image](https://user-images.githubusercontent.com/5755132/104095238-cc025900-52bb-11eb-8d74-d255e322cdcf.png)

8. You will see something like this
![image](https://user-images.githubusercontent.com/5755132/104095248-dfadbf80-52bb-11eb-8391-cb2e08be15fe.png)

9. Select 'sanskrit-lexicon/csl-orig' inside 'head repository'.
![image](https://user-images.githubusercontent.com/5755132/104095295-15eb3f00-52bc-11eb-9947-db85eba30eac.png)

10. You will see the three commits which drdhaval2785/csl-orig was behind sanskrit-lexicon/csl-orig. Click on 'Create Pull Request'
![image](https://user-images.githubusercontent.com/5755132/104095360-5ba80780-52bc-11eb-92da-cca0be4bd3b0.png)

11. Write some commit message in place 1 (I wrote blah blah). Click on 'Create Pull Request' in place 2.
![image](https://user-images.githubusercontent.com/5755132/104095407-a4f85700-52bc-11eb-8eae-414db03d3580.png)

12. Click on 'Merge Pull Request' 
![image](https://user-images.githubusercontent.com/5755132/104095467-f56fb480-52bc-11eb-8f8a-6f289173a2cf.png)

13. Click on 'Confirm Merge'
![image](https://user-images.githubusercontent.com/5755132/104095478-0c160b80-52bd-11eb-8948-9be0db5f5255.png)

14. The changes at sanskrit-lexicon/csl-orig are now pulled to drdhaval2785/csl-orig
![image](https://user-images.githubusercontent.com/5755132/104095502-2c45ca80-52bd-11eb-850a-64580fe46503.png)

15. Go to https://github.com/drdhaval2785/csl-orig and check. Your fork will not be behind any commit. It will be ahead by some commits.
![image](https://user-images.githubusercontent.com/5755132/104095530-57301e80-52bd-11eb-88b9-2198f5b463db.png)

Now your fork is updated. 

You can now create a pull request to sanskrit-lexicon/csl-orig (for change in pe.txt) by following steps mentioned in this comment, as usual.

**Comment by drdhaval2785** [comment]
# Update your local files

Because you pulled some changes from sanskrit-lexicon/csl-orig to drdhaval2785/csl-orig, your local files would not have those changes. The changes are still inside github website only. You need to bring those changes to your files on your computer.
Kindly follow the steps mentioned in https://github.com/sanskrit-lexicon/csl-corrections/issues/31#issuecomment-754444135 to update your local files too.

**Comment by AnnaRybakovaT** [comment]
Dear Dhaval,
Thanks a lot for your instruction! Now I managed to update the forked repo (inside github website and on my computer).

**Comment by funderburkjim** [comment]
>  I managed to update the forked repo

Kudos!
